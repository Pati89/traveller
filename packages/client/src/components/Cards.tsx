import type { FC } from 'react'
import { Grid, GridItem } from '@chakra-ui/react'
import { Card } from './Card'
import type { City } from '../types'

type Cards = {
  cities: City[]
  toggleVisited: (city: City) => Promise<void>
  toggleFavorites: (city: City) => Promise<void>
}

export const Cards: FC<Cards> = ({ cities, toggleVisited, toggleFavorites }) => (
  <Grid templateColumns="repeat(4, 1fr)" gap={6} data-testid="cards">
    {cities.length !== 0
      ? cities.map((city: City) => (
          <GridItem w="100%" minW="250px" maxW="270px" h="100" key={city.id}>
            <Card
              updateVisited={() => toggleVisited(city)}
              updateFavorites={() => toggleFavorites(city)}
              title={city.name}
              country={city.country}
              visited={city.visited}
              favorite={city.wishlist}
            />
          </GridItem>
        ))
      : 'no items to show'}
  </Grid>
)
