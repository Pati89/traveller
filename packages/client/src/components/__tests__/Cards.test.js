import { Cards } from '../Cards'
import { render, screen } from '@testing-library/react'

const props = {
  cities: [
    {
      id: 1,
      title: 'test title 1',
      country: 'country 1',
      visited: false,
      favorite: false,
      updateVisited: () => {},
      updateFavorites: () => {},
    },
    {
      id: 2,
      title: 'test title 2',
      country: 'country 2',
      visited: false,
      favorite: false,
      updateVisited: () => {},
      updateFavorites: () => {},
    },
  ],
  toggleVisited: () => {},
  toggleFavorites: () => {},
}

describe('Cards', () => {
  test('should render a Cards component', () => {
    render(<Cards {...props} />)
    const cards = screen.getByTestId('cards')
    expect(cards).toBeTruthy()
  })

  test('should render a card list', () => {
    render(<Cards {...props} />)
    const card = screen.getAllByTestId('card')
    expect(card).toHaveLength(2)
  })

  test('should not render a card list', () => {
    render(<Cards {...props} cities={[]} />)
    const card = screen.getByText('no items to show')
    expect(card).toBeTruthy()
  })

  test('should render a Cards match snapshot', () => {
    render(<Cards {...props} />)
    const cards = screen.getByTestId('cards')
    expect(cards).toMatchSnapshot()
  })
})
