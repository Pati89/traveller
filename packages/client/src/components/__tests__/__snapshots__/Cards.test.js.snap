// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should render a Cards match snapshot 1`] = `
<div
  class="css-15n0lfp"
  data-testid="cards"
>
  <div
    class="css-1in26ok"
  >
    <div
      class="css-1ggrfbi"
      data-testid="card"
    >
      <h2
        class="css-1dv4iqg"
      >
        , 
        country 1
      </h2>
      <div
        class="css-70qvj9"
      >
        <div
          class="css-q9rcy5"
        >
          <button
            class="chakra-button css-5c2176"
            data-testid="visited-button"
            type="button"
          >
            Not visited
          </button>
        </div>
        <div
          class="css-17xejub"
        />
        <div
          class="css-q9rcy5"
        >
          <button
            aria-label="Send email"
            class="chakra-button css-198bfjk"
            data-testid="favorite-button"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8 2.748l-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 01.176-.17C12.72-3.042 23.333 4.867 8 15z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-1in26ok"
  >
    <div
      class="css-1ggrfbi"
      data-testid="card"
    >
      <h2
        class="css-1dv4iqg"
      >
        , 
        country 2
      </h2>
      <div
        class="css-70qvj9"
      >
        <div
          class="css-q9rcy5"
        >
          <button
            class="chakra-button css-5c2176"
            data-testid="visited-button"
            type="button"
          >
            Not visited
          </button>
        </div>
        <div
          class="css-17xejub"
        />
        <div
          class="css-q9rcy5"
        >
          <button
            aria-label="Send email"
            class="chakra-button css-198bfjk"
            data-testid="favorite-button"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8 2.748l-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 01.176-.17C12.72-3.042 23.333 4.867 8 15z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
